debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -62,18 +63,25 @@
     private static final Duration PAUSE_BETWEEN_REPLICATION_SLOT_RETRIEVAL_ATTEMPTS = Duration.ofSeconds(2);
 
     private final TypeRegistry typeRegistry;
+    private final PostgresDefaultValueConverter defaultValueConverter;
 
     /**
      * Creates a Postgres connection using the supplied configuration.
      * If necessary this connection is able to resolve data type mappings.
      * Usually only one such connection per connector is needed.
      *
      * @param config {@link Configuration} instance, may not be null.
-     * @param provideTypeRegistry {@code true} if type registry should be created
+     * @param typeRegistry {@link TypeRegistry} the type registry to bind to this connection.
+     * @param valueConverter {@link PostgresValueConverter} a postgres value converter.
      */
-    public PostgresConnection(Configuration config, boolean provideTypeRegistry) {
+    public PostgresConnection(Configuration config, TypeRegistry typeRegistry, PostgresValueConverter valueConverter) {
         super(config, FACTORY, PostgresConnection::validateServerVersion, PostgresConnection::defaultSettings);
-        this.typeRegistry = provideTypeRegistry ? new TypeRegistry(this) : null;
```

Comment:
I had a heck of a time trying to tease apart the dependencies between `PostgresConnection`, `TypeRegistry`, and `PostgresValueConverter`.  Open to alternative suggestions!